doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
+======
+
+- PEP8 ignore E501
+- prefer % over format
+- try to avoid generators and decorators
```

Comment:
we could also ban for loops inside for loops. I always lose time trying to understand what the coder meant. Also, for some reason, they are always bugged